diff --git a/dist/mainnet/index.js b/dist/mainnet/index.js
index 656e87042e6a22717ef17ca2fc451e3c1c19daf1..95685550a930f04b04ff58556e18fe69fbff0b8d 100644
--- a/dist/mainnet/index.js
+++ b/dist/mainnet/index.js
@@ -19721,8 +19721,11 @@ async function __wbg_init(module_or_path, memory) {
 }
 
 const module$1 = new URL("aleo_wasm.wasm", import.meta.url);
-                
-                    await __wbg_init({ module_or_path: module$1 });
+const shouldAutoInit = (typeof window !== "undefined" && typeof document !== "undefined") || (typeof WorkerGlobalScope !== "undefined" && typeof self !== "undefined" && self instanceof WorkerGlobalScope);
+
+if (shouldAutoInit) {
+    await __wbg_init({ module_or_path: module$1 });
+}
 
 async function initThreadPool(threads) {
     if (threads == null) {
diff --git a/dist/testnet/index.js b/dist/testnet/index.js
index cca66e1fcf1c3cda22fcdd4774c06b1d01aab33e..930498e5f6860fec48f24e144b38bea9f450468e 100644
--- a/dist/testnet/index.js
+++ b/dist/testnet/index.js
@@ -19721,8 +19721,11 @@ async function __wbg_init(module_or_path, memory) {
 }
 
 const module$1 = new URL("aleo_wasm.wasm", import.meta.url);
-                
-                    await __wbg_init({ module_or_path: module$1 });
+const shouldAutoInit = (typeof window !== "undefined" && typeof document !== "undefined") || (typeof WorkerGlobalScope !== "undefined" && typeof self !== "undefined" && self instanceof WorkerGlobalScope);
+
+if (shouldAutoInit) {
+    await __wbg_init({ module_or_path: module$1 });
+}
 
 async function initThreadPool(threads) {
     if (threads == null) {
